tests/luau/compile.luau

```diff
@@ -0,0 +1,8 @@
+local luau = require("@lune/luau")
+
+assert(type(luau.compile) == "function", "expected `luau.compile` to be a function")
+
+assert(
```

Comment:
Is there any way to verify that this is actually bytecode and not an arbitrary string? Does Luau bytecode start with any specific magic header or somesuch?

Reply:
From what I can tell - Luau bytecode does start off with the version of the bytecode, but the bytes after that go into the content of that bytecode.

<bytecode-ver> <string-tbl> <proto-tbl>

I've found this from both of these sources;

- https://github.com/uniquadev/LuauVM/blob/master/VM/luau/lvmload.lua
- https://github.com/Roblox/luau/blob/bd229816c0a82a8590395416c81c333087f541fd/VM/src/lvmload.cpp

Reply:
> Additionally, if a compilation fails - instead of producing an error, the luau.compile function will return a message where the first byte is 0, and the error is attached afterwards

I think it would probably be a good idea to check for this on the rust end of things for the `compile` function and throw a lua error then, too. We don't really have any other APIs that behave like this in Lune where they return an error message, if the user wants to handle errors/fallible functions `pcall` should be the go-to for consistency

Reply:
I've made the above changes in https://github.com/filiptibell/lune/pull/82/commits/b5906b2ec8657575cc4c0e9c2ec4f768454504d5